docs/topics/tasks.txt

```diff
@@ -304,33 +297,35 @@ conversion.
 Transactions
 ------------
 
-By default, Tasks are enqueued after the current database transaction (if there
-is one) commits successfully (using :meth:`transaction.on_commit()
-<django.db.transaction.on_commit>`), rather than enqueueing immediately. For
-most backends, Tasks are run in a separate process, using a different database
-connection. Without waiting for the transaction to commit, workers could start
-to process a Task which uses objects it can't access yet.
-
-This behavior can be changed by changing the :setting:`TASKS-ENQUEUE_ON_COMMIT`
-setting for the backend, or for a specific Task using the ``enqueue_on_commit``
-parameter.
+For most backends, Tasks are run in a separate process, using a different
+database connection. When using a transaction, without waiting for it to
+commit, workers could start to process a Task which uses objects it can't
+access yet.
 
 For example, consider this simplified example::
 
     @task
-    def my_task():
-        Thing.objects.get()
+    def my_task(thing_num):
+        Thing.objects.get(num=thing_num)
 
 
     with transaction.atomic():
-        Thing.objects.create()
-        my_task.enqueue()
+        Thing.objects.create(num=1)
+        my_task.enqueue(thing_num=1)
 
+To prevent the scenario where ``my_task`` runs before the ``Thing`` is
+committed to the database, use :func:`transaction.on_commit()
+<django.db.transaction.on_commit>`, binding all arguments to
+:meth:`~django.tasks.Task.enqueue` via :func:`functools.partial`::
 
-If :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>` is ``False``, then it
-is possible for ``my_task`` to run before the ``Thing`` is committed to the
-database, and the Task won't be able to see the created object within your
-transaction.
+    from functools import partial
+
+    from django.db import transaction
+
+
+    with transaction.atomic():
+        Thing.objects.create(num=1)
+        transaction.on_commit(partial(my_task.enqueue, thing_num=1))
```

Comment:
Thought: Should there be a test case to confirm this behaviour works? I can't see a reason it wouldn't, but might as well add a test to confirm.

Reply:
@RealOrangeOne That's a great idea. Could you work on that in the incoming days? We'll merge this as is because we can't afford another push and long wait time for the CI run, but I will be thrilled to review and merge that before the beta.

Reply:
@nessita Yep sounds good to me!